app/src/main/java/com/eventyay/organizer/core/event/dashboard/EventDashboardPresenter.java

```diff
@@ -144,7 +145,13 @@ public void toggleState() {
                         getView().onSuccess(utilModel.getResourceString(R.string.draft_success));
                     }
                 },
-                throwable -> event.state = Event.STATE_DRAFT.equals(event.state) ? Event.STATE_PUBLISHED : Event.STATE_DRAFT);
+                throwable -> {
+                    if (ErrorUtils.getErrorDetails(throwable).toString().contains("starts-at should be after current date-time"))
```

Comment:
Check the source of error, not the entire string

Reply:
Ok, also the tests are failing. Can you please guide why? I think the tests should not have required any changes.

Reply:
> Check the source of error, not the entire string

Updated.

Reply:
https://circleci.com/gh/fossasia/open-event-organizer-android/3385?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Reply:
https://3385-65606855-gh.circle-artifacts.com/0/reports/tests/testPlayStoreDebugUnitTest/index.html

Reply:
@iamareebjamal Any help on why the tests aren't passing? I have discussed with @anhanh11001 but he is also not able to figure out the reason.